source/simulator/core.cc

```diff
@@ -2241,6 +2256,297 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Newton_Stokes:
+        {
+          if (parameters.free_surface_enabled)
```

Comment:
I checked it with the crustal 2d cookbook in the sense that it converges (for the first few timesteps) and creates a mountain as you would expect. I have convergence problems after the first few timesteps (after about 12 timesteps) and I noticed that the iterated Stokes was sometimes outperforming the defect corrected Picard significantly, but I don't think this has anything to do with the free surface, because removing the free surface and leaving the top surface open yields a similar result. It might just be that the Drucker Prager is very difficult, or there is something wrong with an open top boundary, although in the Spiegelman benchmark which is very similar in the setup does not show this behaviour (even the opposite).